Zend/zend_globals.h

```diff
@@ -295,6 +295,13 @@ struct _zend_executor_globals {
 	zend_string *filename_override;
 	zend_long lineno_override;
 
+	/* Guards are context dependent. I.e. if __get() is being called for an object
+	 * within a fiber, the guard will _not_ skip __get() in the main context. To
+	 * achieve this, we offset the guard string hash by the guard context.
+	 * Additionally, bailout will discard the current guards in the same way. */
+	uint32_t guard_context;
+	uint32_t guard_context_counter;
```

Comment:
I'm quite afraid of 32 bit integer overflows for long running scripts here. It's a very good source of a heisenbug, especially if the counter overflows to 0 and a fiber and a main thread happen to suspend at the same place.

Reply:
Sounds reasonable 👍